app/models/mixins/retirement_mixin.rb

```diff
@@ -26,6 +27,13 @@ def retire(ids, options = {})
       q_options.merge!(:user_id => user.id, :group_id => user.current_group.id, :tenant_id => user.current_tenant.id) if user
       MiqQueue.put(q_options)
     end
+
+    def set_retirement_requester(obj_name, obj_ids, requester)
+      obj_ids.each do |id|
```

Comment:
This will be doing an N+1 for looking up object_ids. Not sure how many there will be. Or if some of the ids will be invalid.

```ruby
obj_name.constatize.where(:id => obj_ids).each do |object|
  _log.info("Retirement requester for #{obj_name} #{object.id} being set to #{requester.userid}")
  object.update_attributes(:retirement_requester => requester.userid))
end
```

- Not sure if you need to count the query result and raise a not found error or something
- Also, can't remember the way we typically display `class:id`, but thought it had `[]` or something.
- an `update_all(:retirement_requester => requester.userid)` would just do it in the database and not fetch any of the results - again, an optimization that may make sense or not.


Or